### DISCUSSION TITLE
How to install Keystone (from git) for dev, contributing, or testing purposes?

### DISCUSSION THREAD

**Discussion by anthonyivol** [keystonejs/keystone#7794] [Questions] [answered]
I've found in the CONTRIBUTING.MD a link to a monorepo style guide but nothing on how to setup **this** project globally.

I tried a git clone of a fresh fork, outside of my project node_modules, yarn link everything from the packages directory, `keystone postinstall` & `yarn dev` but got : 

![image](https://user-images.githubusercontent.com/19983700/184548245-bb1067bc-b8e0-4535-9fcd-774e0f2de239.png)

I also tried to install the fork with `yarn add user/keystone.git`, `cd node_modules/@keystone-6/mono-repo/packages/core`, `yarn link`, `cd node_modules/@keystone-6/mono-repo`, `yarn prepare` and got :

![image](https://user-images.githubusercontent.com/19983700/184548251-53e30d5a-db9c-4f81-a1a7-ea3d914a7d47.png)

So what is the correct way to install a fork of this repo as a project dependency ?

**Comment by Noviny** [comment, marked as answer]
Hey @anthonyivol - we don't have a clear-stated path on how we'd recommend you do this. I just looked into how I would best do this, and for things that require being global, you want to symlink between keystone's `node_modules` into the `node_modules` of your own project.

After you have symlinked the keystone package, run the following in your project:

```shell
rm -rf node_modules/graphql
ln -s /{FULL_PATH_TO_KEYSTONE_CLONE}/keystone/node_modules/graphql node_modules/graphql
```

This should get you to using the same versions of `graphql` in both places so you can test your changes. This is probably the best way forward.

**Reply by anthonyivol** [reply]
Thanks, in addition I had to link also react, react-dom and next.
It definitely should be something in CONTRIBUTING.md more concise on how to proceed.

**Reply by dcousens** [reply]
@anthonyivol we'd love if you wanted to make a pull request adding that information :heart_eyes:

**Reply by anthonyivol** [reply]
Done : https://github.com/keystonejs/keystone/pull/7796 ✌️
